engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -101,13 +104,13 @@ public T getValue() {
     }
 
     public boolean setValue(T newValue) {
-        if (!validator.validate(newValue))
+        if (!validator.validate(newValue)) {
+            LOGGER.warn("The passed value {} is invalid.", newValue);
```

Comment:
In my earlier comment I thought of suggesting option 1, but then I thought you'd figure it out yourself... which you did! :smiley: 

If a developer is setting a value and the operation return false the developer will wonder: why was I not able to set the value? As such, I'd explain in the javadoc of the method to check the log for the exact nature of the problem when the method returns false.